src/Google/RequestReviewStatuses.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Google;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+
+/**
+ * Helper class for Account request Review feature
+ */
+class RequestReviewStatuses implements Service {
+
+	public const DISAPPROVED    = 'DISAPPROVED';
+	public const WARNING        = 'WARNING';
+	public const ELIGIBLE       = 'ELIGIBLE';
+	public const UNDER_REVIEW   = 'UNDER_REVIEW';
+	public const PENDING_REVIEW = 'PENDING_REVIEW';
+	public const ONBOARDING     = 'ONBOARDING';
+
+	/**
+	 * Merges the different program statuses based on priority
+	 *
+	 * @param array $response
+	 *
+	 * @return array
+	 */
+	public function get_statuses_from_response( array $response ) {
+		$issues         = [];
+		$review_allowed = false;
+		$status         = 'APPROVED';
+
+		foreach ( $response as $program_type ) {
+			foreach ( $program_type['data']['regionStatuses'] as $region_status ) {
+
+				if ( $region_status['reviewEligibilityStatus'] === $this::ELIGIBLE ) {
+					$review_allowed = true;
+				}
+
+				if ( $region_status['eligibilityStatus'] === $this::DISAPPROVED || $region_status['eligibilityStatus'] === $this::WARNING ) {
+					if ( $status !== $this::DISAPPROVED ) {
+						$status = $region_status['eligibilityStatus'];
+					}
+
+					$issues = array_merge( $issues, $region_status['reviewIssues'] ?? [] );
+				}
+
+				if ( $status === $this::DISAPPROVED ) {
+					continue;
+				}
+
+				if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
```

Comment:
In cases where the $status variable is checked after an assignment wouldn't be better to immediately `continue` after the assignment?

		```
	if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
					$status = $this::UNDER_REVIEW;
				}

				if ( $status === $this::UNDER_REVIEW ) {
					continue;
				}
```
could be:
```

			if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
					$status = $this::UNDER_REVIEW;
					continue;
				}

```

Reply:
Not sure about that @rawdreeg 

**Right now:**

First iteration -> region US:  eligibilityStatus =  UNDER_REVIEW

```
if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
	$status = $this::UNDER_REVIEW; // -> Reached --> $status = $this::UNDER_REVIEW
}

if ( $status === $this::UNDER_REVIEW ) {
	continue; // --> Reached 
}
```

Second iteration -> region MX:  eligibilityStatus =  PENDING_REVIEW

```
if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
	$status = $this::UNDER_REVIEW; // -> NOT  Reached
}

if ( $status === $this::UNDER_REVIEW ) {
	continue; // --> Reached 
}

if ( $region_status['eligibilityStatus'] === $this::PENDING_REVIEW ) {
	$status = $this::PENDING_REVIEW; // NOT REACHED
}

```
Hence the result is correct because UNDER_REVIEW as priority over PENDING_REVIEW

**Proposed change:**

First iteration -> region US:  eligibilityStatus =  UNDER_REVIEW
```
if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
	$status = $this::UNDER_REVIEW; // -> Reached --> $status = $this::UNDER_REVIEW
	continue; // --> Reached 
}

```

Second iteration -> region MX:  eligibilityStatus =  PENDING_REVIEW

```
if ( $region_status['eligibilityStatus'] === $this::UNDER_REVIEW ) {
	$status = $this::UNDER_REVIEW; // -> NOT  Reached
	continue; // --> NOT Reached 
}

if ( $region_status['eligibilityStatus'] === $this::PENDING_REVIEW ) {
	$status = $this::PENDING_REVIEW; // REACHED -> STATUS -> PENDING_REVIEW
}
```
Hence the result is not correct because UNDER_REVIEW as priority over PENDING_REVIEW and we evaluated status to PENDING_REVIEW

Reply:
Thanks for clarifying.  I initially missed that  `$status ` needed to be checked in subsequent iterations.